### ISSUE TITLE
[question] How to access conf variables for the build context in the generate method?

### ISSUE THREAD

**Issue by jwillikers** [conan-io/conan#12860]
### What is your question?

The [conf attribute](https://docs.conan.io/en/latest/reference/conanfile/attributes.html#conf) may contain settings for the build context which recipes need, such as `compiler_executables`. The `libcap` recipe uses `BUILD_CC` environment variable, see [here](https://github.com/conan-io/conan-center-index/blob/master/recipes/libcap/all/conanfile.py#L66-L69), to configure the C compiler to use in the build context. I don't see a way to access `compiler_executables` for the build context to properly set this environment variable in the `generate` method.

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @jwillikers 

If the ``tool_requires`` define any value in its ``self.conf_info``, then its direct consumers will automatically have it available in the ``self.conf``, so directly reading ``self.conf.get("tools.build:compiler_executables")`` should be enough, have you tried it?

**Comment by jwillikers** [comment]
@memsharded Sorry, I must not have been clear. I mean that I need to access the conf value defined in the _profile_ for the build context instead of the host context. If `tools.build:compiler_executables` is defined in a profile used for `--profile:build`, how do I access its value in a recipe? `self.conf.get()` would give me the `tools.build:compiler_executables` configured in the host profile, right?

**Comment by memsharded** [comment]
> @memsharded Sorry, I must not have been clear. I mean that I need to access the conf value defined in the profile for the build context instead of the host context. If tools.build:compiler_executables is defined in a profile used for --profile:build, how do I access its value in a recipe? self.conf.get() would give me the tools.build:compiler_executables configured in the host profile, right?

Not really. ``self.conf.get()`` will give you the configuration applied to the current recipe. If the recipe is in the "host" context, it will automatically get it from the "host" profile, but if the package is in the "build" context it will automatically get the ``conf`` from the "build" context.

**Comment by jwillikers** [comment]
The `libcap` recipe uses Makefiles which use the value `BUILD_CC` for the build machine's compiler. `CC` is used as usual for the compiler in the host context. So when building the recipe in the host context, its not possible to set this to the value provided for the build context through `tools.build:compiler_executables`?

**Comment by memsharded** [comment]
Conan now uses internally a ``conf_build`` to point to the build context, and that is used internally by MesonToolchain when necessary to access different conf like ``tools.build:compiler_executables`` or ``tools.build:sysroot`` when ``native=True``. And by ``AutotoolsToolchain`` to define ``CC_FOR_BUILD`` variables.

Maybe this ticket could be closed now?
